scripts/mk/go-rules.mk

```
@@ -119,6 +129,12 @@ $(patsubst cmd/%,$(BIN)/%,$(wildcard cmd/*)): $(shell find $(PROJECT_DIR)/cmd -t
 # $(shell find "$(EVENT_MESSAGE_DIR)" -type f -name '*.go'): $(SCHEMA_YAML_FILES)
 # 	$(MAKE) gen-event-messages
 
+.PHONY: pr-check
+pr-check: ## Run common checks before submitting a PR
+	$(MAKE) go-fmt
+	$(MAKE) container-tidy vet
+	$(MAKE) generate-api generate-mock install-go-tools
+	$(MAKE) build test-unit
```

I like the idea of `pr-check` rule; I suggest the below:

```suggestion
pr-check: ## Run common checks before submitting a PR
	[ ! -e "$(PROJECT_DIR)/tools/bin" ] || rm -rvf "$(PROJECT_DIR)/tools/bin"
	$(MAKE) -j1 compose-clean clean install-tools build lint compose-up test container-build
```

* vet is actually invoked by `make lint` (it is enabled on `.golangci.yaml` file). Indeed `make vet` rule could be removed.
* for the formatting it is setup `gofumpt` see: `.vscode/settings.json` and `docs/CONTRIBUTING.md` files.

---

- `make lint` uses pre-commit. I don't particularly like the tool, because it is too magic to debug and yet another layer of indirection.
- I prefer `pr-check` to be lightweight and not involve compose and container stuff.

---

I don't full agree; I think in freeipa-operator repository I made the mistake of adding many linters and validations (no leveraging pre-commit) on the repository and pipeline because I didn't know the potential of pre-commit.

After that learning, I have tried to delegate as much as possible on the pre-commit configuration (using or not as a hook for git-commit, which I leave to the decision of each developer); the reason is because it is pretty comfortable to add new linters from existing hooks and no reinvent the wheel; everything is installed by `make install-tools` as indicated on the getting started doc (once task).

I agree on extract `compose-clean` and `compose-up` from the proposed rule as that would make it incompatible with the execution of the `make pr-check` from the github pipeline, delegating on the developer to be aware that it should be started the local infrastructure before run the rule (be aware `make test` now execute `test-unit` and `test-smoke` rules for the matching type of tests, and `test-smoke` require the local infrastructure).

Said that, accepting the changes and it can always be changed in the future if it were necessary.